source/common/http/conn_manager_impl.cc

```diff
@@ -451,6 +464,10 @@ void ConnectionManagerImpl::onEvent(Network::ConnectionEvent event) {
     user_agent_.onConnectionDestroy(event, true);
     resetAllStreams(absl::nullopt);
   }
+
+  if (close_type.has_value()) {
+    read_callbacks_->connection().close(close_type.value());
```

Comment:
I'm a bit worried about the resetAllStreams with regards to the new reentrant nature.  

I was digging through code to see if resetAllStreams wrote stream resets to the wire (which I assume it does if we're doing a locally motivated close, but you'd probably know offhand), and I'm worried about a scenario where we're say handling a codec exception, we decide to reset streams, we hit frame flooding limits, and throw another exception.  I guess given we already called resetStreams this refactor doesn't make that scenario worse, but the reentrant nature has me thinking of corner cases I'd like to make sure we're good against.

Reply:
`resetAllStreams` doesn't write anything to the wire. It just pretends that all the streams got remote reset so that everything upstream is cleaned up, so I think this is safe. Let me add more comments.